### PR TITLE
openstack/nfv: enable SR-IOV testing on mellanox

### PR DESCRIPTION
**Pull request by EmilienM**
* Remove `openshift-e2e-openstack-dpdk` workflow, we don't need it
  anymore because we can now test both DPDK & SRIOV on Mellanox
  networks.

* Therefore, remove `openstack-provision-scale-worker` step, which isn't
  used anymore.

* Update the jobs for 4.11, 4.10 and 4.9 to test SRIOV with Mellanox
  (the regular `openshift-e2e-openstack-nfv` workflow).

* A few changes in `sriov-networknodepolicy`: set `isRdma` to `True` if
  Mellanox and configure the `deviceType` to `netdevice` (both are
  required).

* For the `test-sriov` step, we need to patch
  `network-attachment-definitions`, so testpmd can use the Mellanox VF
  device. This is not needed for Intel.

### CODE REVIEW

File: ci-operator/step-registry/openstack/provision/sriov-networknodepolicy/openstack-provision-sriov-networknodepolicy-commands.sh [hunk 1]
```diff
@@ -54,12 +56,20 @@ then
...
+else
+    SRIOV_DEVICE_TYPE="vfio-pci"
+    IS_RDMA="false"
+fi
+
+create_sriov_networknodepolicy "sriov1" "${OPENSTACK_SRIOV_NETWORK}" "${SRIOV_DEVICE_TYPE}" "${IS_RDMA}"
 
 if [[ "${OPENSTACK_DPDK_NETWORK}" != "" ]]; then
     if oc get MachineConfig/99-vhostuser-bind >/dev/null 2>&1; then
         echo "vhostuser is already bound to the ${OPENSTACK_DPDK_NETWORK} network."
```
Comment (MaysaMacedo) [inline ci-operator/step-registry/openstack/provision/sriov-networknodepolicy/openstack-provision-sriov-networknodepolicy-commands.sh:71]: Just to confirm, this is harmless right?
I saw that the sriov node policy reported this.

**Comment by MaysaMacedo** [comment]
/lgtm
/hold unhold to merge it 

The CI failure was just a timeout.

**Comment by EmilienM** [comment]
/retest
making sure it works fine on 4.9 too.

File: ci-operator/step-registry/openstack/provision/sriov-networknodepolicy/openstack-provision-sriov-networknodepolicy-commands.sh (hunk 1 above)
Comment (EmilienM) [reply ci-operator/step-registry/openstack/provision/sriov-networknodepolicy/openstack-provision-sriov-networknodepolicy-commands.sh:135]: yeah, if you saw this in the logs it means that it's a job that deploys the `vhostuser` step registry and will create a network policy for DPDK.

**Pull request merged**